### ISSUE TITLE
Noise reduction works, than it doesn't.

### ISSUE THREAD

**Issue by quamt** [rigaya/QSVEnc#185]
There is strange behaviour with QSVEnc when using `--vpp-convolution3d` and batch encoding. 

In the first video that gets encoded, the noise/grain gets removed. 
The noise/grain gets enhanced/multiplied strangely in each following video. 

I need help figuring out where the issue is coming from.
When I encode them individually, the problem does not occur.
I use a Windows batch script for QSVEnc operations.

**Comment by rigaya** [comment]
> When I encode them individually, the problem does not occur.

Then there should be something wrong with the bat script.

I don't have such strange behavior, in general each process will not be affected by the other (previous) process.

**Comment by quamt** [comment]
The batch is a simple `for-loop`. That shouldn't be the problem, as I have been using `for-loops` since long ago.

However, I noticed the issue could be related to my multi-GPU setup and usage of the `--device option`.

When I remove the option `--device 1` or `--device 2` from my command and let QSVEnc select the device, the `--vpp-convolution3d` option results looks ok, I tested 20 grainy files and trimm 5000 frames windows for the testing, as I always do when testing filters.

I'm very surprised that it only happens with `--vpp-convolution3d` or maybe I didn't notice it with the other noise removal filters as I don't use them so often.

**Comment by quamt** [comment]
@rigaya 

I did some more testing, and it pinpointed that option when multiple GPUs are used.
I tested running the batch and let only one GPU do the job.
As soon as `--device` is removed, the issue disappears. 
Are there any logs I can provide that might reveal anything?

**Comment by quamt** [comment]
Ok. I was able to determine the culprit of the issues.
I am using a multi-GPU setup for my testing, including the iGPU.
1st, I switched off the iGPU to see if that solved the problem, but the issue persisted. 
Then I remembered that I activated the Graphic Preference in Windows > System > Display > Graphics > Selected App (QSVenc).
I removed that option since I also deactivated the iGPU. 
Now, the processing seems to perform as intended. 
I will conduct some further tests and report back.
Thank you.
